flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/split/BigQuerySourceSplit.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.source.split;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.connector.source.SourceSplit;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/** A {@link SourceSplit} implementation for a BigQuery Read API stream. */
+@PublicEvolving
+public class BigQuerySourceSplit implements SourceSplit, Serializable {
+
+    private final String streamName;
+    private final Integer offset;
+
+    public BigQuerySourceSplit(String streamName) {
+        this.streamName = streamName;
+        this.offset = 0;
+    }
+
+    public BigQuerySourceSplit(String streamName, Integer offset) {
+        this.streamName = streamName;
+        this.offset = offset;
+    }
+
+    @Override
+    public String splitId() {
```

Comment:
each stream will only have one split? As splitId needs to be unique with the scope of the source.

Reply:
That's correct, every Flink split correlates with a read stream.

Reply:
I may misunderstand the concept here. What does a `stream` mean here? If it represents a data source, then we should have multiple splits under one stream, right?

Reply:
So a Flink split is a logical partition the source data we intent to read. When implementing a Source we need to make a decision on how that logical representation of a data portion will map with the underlying system. 

In the case of BigQuery, when using the StorageRead API the data can be read in a massively parallel way natively since the API partition the underlying data in a bunch of ReadStreams, each of those will be represented from the client library perspective as a remote iterator to which we can request a bunch of encoded rows (this can be in Avro or Arrow format). 

Since BQ already has a clear concept that logically partition the data that can be read from the storage source (a table or a table partition), we can use it to directly represent a Flink Split. 

So for this implementation we can minimally say Flink Split == BQ ReadStream. 

This will terminally mean that when reading a full BQ table or partition, we will be splitting the read in N different portions of data, each of them correlating with one BQ ReadStream following the natural definition of the client library.

Reply:
Gotcha. Thanks for the explanation.